### PR TITLE
{WIP} Image search - fetching all image type files

### PR DESCRIPTION
**Pull request by pritish-devurkar**
## Purpose

This PR adds support to image search to the KM App.

Currently fetches from only image type files, embedded images coming in subsequent commits

## Does this introduce a breaking change?

```
[ ] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: Web UI/CognitiveSearch.UI/Controllers/HomeController.cs [hunk 1]
```diff
@@ -65,6 +65,23 @@ public IActionResult Search(string q)
 
             return View();
         }
+        public IActionResult Images()
+        {
+            return View();
+        }
+        [HttpPost, HttpGet]
+        public IActionResult Images(string q)
```
Comment (grecoe) [inline Web UI/CognitiveSearch.UI/Controllers/HomeController.cs:73]: What happens if the incoming string is empty? Could there be any bad side effects with injection or enforcing ownership/rights to a query?

File: Web UI/CognitiveSearch.UI/wwwroot/js/results.js [hunk 2]
```diff
@@ -428,4 +428,76 @@ function UpdateResults(data) {
...
+                            <div class="search-result">
+                                <img class="img-result"  style='max-width:100%;' src="${path}"/>';
+                                    <div class="results-header">
+                                    <h4>${name}</h4>
+                                    </div>
+                                </div>
+                            </div>
+                            </div>`;
+        }
+                // resultsHtml += '    <div class="image-result-path" >';
```
Comment (grecoe) [inline Web UI/CognitiveSearch.UI/wwwroot/js/results.js:478]: Remove unused code

**Review by grecoe — APPROVED** [review]
Would really like to see comments in the code explaining what a function is used for. Also noted several areas where dead code should be removed. Otherwise, not 100% sure of the execution so going to accept it so we can move it over to One.

File: Web UI/CognitiveSearch.UI/Controllers/HomeController.cs (hunk 1 above)
Comment (pritish-devurkar) [reply Web UI/CognitiveSearch.UI/Controllers/HomeController.cs:73]: Oh no, this part of the code, just deals with logged the searched query term and searchId to application insights.
So, if this incoming string is empty, it should log a blank string as the query term

File: Web UI/CognitiveSearch.UI/wwwroot/js/details.js [hunk 3]
```diff
@@ -375,4 +384,70 @@ function SearchTranscript(searchText) {
         // get whole phrase
         GetReferences(searchText, false);
     }
-}
\ No newline at end of file
+}
+
+// function GetImagesHTML(result) {
```
[the remark this answers is not in the document]
Comment (pritish-devurkar) [reply Web UI/CognitiveSearch.UI/wwwroot/js/details.js]: Removed this.

File: Web UI/CognitiveSearch.UI/wwwroot/js/results.js (hunk 2 above)
Comment (pritish-devurkar) [reply Web UI/CognitiveSearch.UI/wwwroot/js/results.js]: Removed this

**Comment by pritish-devurkar** [comment]
> Would really like to see comments in the code explaining what a function is used for. Also noted several areas where dead code should be removed. Otherwise, not 100% sure of the execution so going to accept it so we can move it over to One.

Thanks @grecoe for your review and comments, I have removed all the commented pieces of the code.
Also, in terms of adding comments, I will make sure to add them when I check-in part 2 of the Image Search functionality.

**Pull request merged**